Add Psalm back to the project and CI

lptn opened the pull request:
I'm a https://github.com/psalm/psalm-plugin-laravel contributor and recently added few important features to this package, so I believe it can find some issues that Larastan couldn't. I see your project used Psalm in the past and this is an attempt to use it again, even more actively as I've integrated it into CI.

lptn commented on app/Http/Controllers/UrlRedirectController.php:
```diff
@@ -22,7 +22,7 @@ public function __construct(
      * @return \Illuminate\Http\RedirectResponse
      *
-     * @throws \Illuminate\Database\Eloquent\ModelNotFoundException<\App\Models\Url>
+     * @throws \Illuminate\Database\Eloquent\ModelNotFoundException
```
ModelNotFoundException is not a generic

realodix commented:
it's finished? 

I see Psalm is reporting lots of errors and it seems that it can't recognize Laravel's helper functions


![noname](https://user-images.githubusercontent.com/1314456/214093082-7bd0575f-b51a-4ddb-9ffc-22007e743f91.png)

lptn commented:
Hey @realodix 
Yes, it's finished. I've checked few times locally - it should not produce these errors as they suppressed by baseline file I've added to this PR.

Can you please confirm you made these steps:
 1. Switched to my branch 
 2. run `composer i`
 3. run `composer psalm`

Thanks!

lptn commented:
@realodix 
I found an issue, it caused different package versions. I've updated laravel-plugin accordingly, should be fixed now
